### PR TITLE
Computing the classification BMA, i.e. average of softmaxes, online.

### PR DESCRIPTION
**Pull request by wiseodd**
@runame let me know if this implementation is good

### CODE REVIEW

File: laplace/baselaplace.py [hunk 1]
```diff
@@ -679,6 +680,15 @@ def _nn_predictive_samples(self, X, n_samples=100, **model_kwargs):
             fs = torch.softmax(fs, dim=-1)
         return fs
 
+    def _nn_bma_clf(self, X, n_samples=100, **model_kwargs):
```
Comment (runame) [inline laplace/baselaplace.py:683]: I would prefer a more explicit name for this function, e.g. `_nn_predictive_classification`.

File: laplace/baselaplace.py [hunk 2]
```diff
@@ -679,6 +680,15 @@ def _nn_predictive_samples(self, X, n_samples=100, **model_kwargs):
...
+        py = 0
+        for sample in self.sample(n_samples):
+            vector_to_parameters(sample, self.params)
+            logits = self.model(X.to(self._device), **model_kwargs).detach()
+            py = py + 1/n_samples * torch.softmax(logits, dim=-1)
```
Comment (runame) [inline laplace/baselaplace.py:688]: Nit: I would prefer `py += torch.softmax(logits, dim=-1) / n_samples`.

**Review by runame — APPROVED** [review]
Besides my comments, LGTM!

File: laplace/baselaplace.py (hunk 2 above)
Comment (wiseodd) [reply laplace/baselaplace.py]: Done!
